org.civicrm.reqangular/src/common/angular-date/src/services/DateFormatFactory.js

```diff
@@ -0,0 +1,36 @@
+/**
+ * This factory provides access to date format from CRM settings
+ * @param $q
+ * @returns {{dateFormat: null, getDateFormat: getDateFormat}}
+ * @constructor
+ */
+function DateFormatFactory($q) {
+    return {
+        /**
+         * keeps information about date format
+         */
+        dateFormat: null,
```

Comment:
Maybe you can put the default date format directly here. If the async call will succeed, then the value of `this.dateFormat` will get replaced, otherwise it will be already defined here instead of getting assigned in the `catch` block later on